components/brave_wallet/browser/keyring_service.cc

```diff
@@ -2564,18 +2587,25 @@ KeyringService::SignMessageByBitcoinKeyring(
   return bitcoin_keyring->SignMessage(key_id, message);
 }
 
-std::vector<mojom::AccountInfoPtr> KeyringService::GetAllAccountInfos() {
-  std::vector<mojom::AccountInfoPtr> account_infos;
-  for (const auto& keyring_id : GetSupportedKeyrings()) {
-    for (auto& account_info : GetAccountInfosForKeyring(keyring_id)) {
-      account_infos.push_back(std::move(account_info));
+void KeyringService::ResetAllAccountInfosCache() {
+  account_info_cache_.reset();
+}
+
+const std::vector<mojom::AccountInfoPtr>& KeyringService::GetAllAccountInfos() {
```

Comment:
On fresh install/profile I'm seeing a nil `selected_account` returned from `GetAllAccountInfos()` when restoring a wallet on iOS on this branch. Creating a wallet always has non-nil `selected_account`; seems specific to restoring wallet on fresh install/profle.

Reply:
Couldn't reproduce anything similar with my desktop setup and this PR doesn't touch account selections. Maybe there is some iOS-specific case. Can we debug it separately? Does it block this PR?